pandas/core/sorting.py

```diff
@@ -431,21 +431,32 @@ def safe_sort(values, labels=None, na_sentinel=-1, assume_unique=False):
 
     def sort_mixed(values):
         # order ints before strings, safe in py3
+        from pandas import Period
```

Comment:
why are you going thru all this trouble? not to mention this is going to be less performant.

let's just check that things are strings or ints and be done (anything else can just return as is)

Reply:
So don't bother sorting non-(str or int)?  OK.

Reply:
There are 4 tests that fail once we start ignoring non-(str or int); they are specifically checking for TypeErrors being raised here.